wicker/core/datasets.py

```diff
@@ -207,6 +255,119 @@ def schema(self) -> DatasetSchema:
                 )
         return self._schema
 
+
+class FileSystemDataset(BaseDataset):
```

Comment:
Make a factory method or builder to avoid repeating the S3Dataset mistake again.

Reply:
@zhenyu one second I'm not finished!

Reply:
@zhenyu I just added it. Could you review the new `DatasetFactory` class? I am happy to make adjustments. I added factory methods for both `create_filesystem_dataset` and `create_s3_dataset`, but we could remove the S3 method if you don't want it. For `create_filesystem_dataset` we could make more adjustments to its parameters and/or the `FileSystemDataset.__init__` function parameters.

Reply:
Since the default for `FileSystemDataset` should be no-cache, let me suggest that a very slight alternative plan!

Let's leave the name of the current class as `FileSystemDataset`. But if we ever have a version or subclass of `FileSystemDataset` that caches files (by copying them from the mounted FUSE root directory into another directory on the instance's local disk), let's give that one a more specific name (like `CacheFileSystemDataset` or `FileSystemCacheDataset`).